python/pyarrow/_compute.pyx

```diff
@@ -2234,7 +2234,13 @@ cdef class Expression(_Weakrefable):
 
         for argument in arguments:
             if not isinstance(argument, Expression):
-                raise TypeError("only other expressions allowed as arguments")
+                # Attempt to help convert this to an expression
+                try:
+                    import pyarrow.compute as pc
+                    argument = pc.scalar(argument)
```

Comment:
`pc.scalar` is basically just a wrapper around `Expression._scalar(..)`, something that is available in this file. So I think you can use that here directly instead of the import

Reply:
Aye, that's better. I should have been able to catch that. Thanks.